File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeStars = function(s) {
    
    let stack = [];
    for ( let char of s){
        if (stack.length > 0 && char === "*"){
            stack.pop()
            }
        else{
            stack.push(char);
            }}
    return stack.join("");


};

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []
        #iterate over the string 
        for curr in s:
            if curr == "*":
                if stack[-1] != "*" and len(stack) != 0:
                    stack.pop()

            else:
                stack.append(curr)

        return "".join(stack) 
